ckanext/canada/plugins.py

```diff
@@ -61,6 +64,82 @@
 fq_portal_release_date_match = re.compile(r"(portal_release_date:\"\[.*\]\")")
 
 
+class CanadaThemePlugin(p.SingletonPlugin):
+
+    p.implements(p.IConfigurer)
+    p.implements(p.ITemplateHelpers)
+
+    # IConfigurer
+    def update_config(self, config):
+        p.toolkit.add_template_directory(config, 'templates')
+        p.toolkit.add_public_directory(config, 'public')
+        p.toolkit.add_resource('public/static/js', 'js')
+        p.toolkit.add_resource('assets/internal', 'canada_internal')
+        p.toolkit.add_resource('assets/datatables', 'canada_datatables')
+        p.toolkit.add_resource('assets/public', 'canada_public')
+        set_app_global('is_registry', plugin_loaded('canada_internal'))
```

Comment:
maybe wouldn't need a separate plugin if you don't call plugin_loaded from inside update_config? You can always check the plugin config instead of using this function.

Reply:
@wardi the the plugin is more for the loading of templates over other plugins.

The `plugin_loaded` should be fine in the `update_config` hook as all the plugins should already be in the `_PLUGINS` at that point in `ckan.plugins.core`. So even if `canada_theme` is loaded first, it should still be able to see the loaded plugins during `update_config`